src/sdk/wire_bytes.rs

```diff
@@ -27,41 +32,62 @@ pub fn encode<T: Serialize>(payload: &T) -> TofnResult<BytesVec> {
     })
 }
 
+/// Serialize a value using bincode and log errors
 pub fn serialize<T: ?Sized>(value: &T) -> TofnResult<BytesVec>
 where
     T: serde::Serialize,
 {
-    match bincode::serialize(value) {
-        Ok(bytes) => Ok(bytes),
-        Err(err) => {
-            error!("serialization failure: {}", err.to_string());
-            Err(TofnFatal)
-        }
-    }
+    // Create serialization options for bincode.
+    // The default options don't bound pre-allocation size,
+    // use little-endian and varint encoding, and reject trailing bytes.
+    let options = DefaultOptions::new()
+        .with_limit(MAX_MSG_LEN)
+        .with_big_endian() // do not ignore extra bytes at the end of the buffer
+        .with_varint_encoding() // saves a lot of space in smaller messages
+        .reject_trailing_bytes(); // do not ignore extra bytes at the end of the buffer
+
+    options.serialize(value).map_err(|err| {
+        error!("serialization failure: {}", err.to_string());
+        TofnFatal
+    })
+}
+
+/// Deserialize bytes to a type using bincode and log errors
+pub fn deserialize<T: DeserializeOwned>(bytes: &[u8]) -> TofnResult<T> {
+    let options = DefaultOptions::new()
+        .with_limit(MAX_MSG_LEN)
+        .with_big_endian()
+        .with_varint_encoding()
+        .reject_trailing_bytes();
+
+    options.deserialize(bytes).map_err(|err| {
+        error!("deserialization failure: {}", err.to_string());
+        TofnFatal
+    })
 }
 
 /// deserialization failures are non-fatal: do not return TofnResult
-pub fn decode<T: DeserializeOwned>(bytes: &[u8]) -> Option<T> {
-    let bytes_versioned: BytesVecVersioned = bincode::deserialize(bytes)
-        .map_err(|err| {
-            warn!("outer deserialization failure: {}", err.to_string());
-        })
-        .ok()?;
+pub fn decode<T: DeserializeOwned>(bytes: &[u8]) -> TofnResult<T> {
```

Comment:
Not sure we should return `TofnResult` here instead of `Option`.  `TofnFatal` is intended for fatal errors only but deserialization failures are non-fatal.

Reply:
Good point. I forgot why we had it as an `Option` type while `serialize` returned `TofnResult`. I don't like the inconsistency between the two, so maybe we should return the bincode error? Let's discuss this over a call.